HOApp/Views/POEntryView.xaml.cs

```diff
@@ -25,21 +28,19 @@ public POEntryView()
 
         private void Insert_Click(object sender, RoutedEventArgs e)
         {
-            POEntryViewModel pvm = (POEntryViewModel)this.DataContext;
-            POlineVM poLine = new POlineVM();
-
-            //insert row into grid
-
-            pvm.PO.POlines.Add(poLine);
+            System.Console.WriteLine("Clicked");
+            ViewModel.PO.AddProductOrderLine(ViewModel.SelectedProduct);
         }
 
         private void Approve_Click(object sender, RoutedEventArgs e)
         {
-            POEntryViewModel pvm = (POEntryViewModel)this.DataContext;
+            using (var db = new RetailDbContext())
+            {
+                db.Pos.Add(ViewModel.PO.TheEntity);
+                db.SaveChanges();
+            }

```

Comment:
Good rule of thumb is that if it implements IDisposable, use it in a using block (`DBContext` in this case)

Reply:
Yes, I should be doing this everywhere